packages/playwright-core/src/client/fetch.ts

```diff
@@ -214,7 +214,7 @@ export class APIResponse implements api.APIResponse {
   }
 
   ok(): boolean {
-    return this._initializer.status === 0 || (this._initializer.status >= 200 && this._initializer.status <= 299);
+    return (this._initializer.status >= 200 && this._initializer.status <= 299);
```

Comment:
add a test with file:// url ?

Reply:
added one. For our api testing we don't support the 'file:' protocol.